finagle-netty4/src/main/scala/com/twitter/finagle/netty4/ssl/client/SslClientVerificationHandler.scala

```diff
@@ -74,14 +76,14 @@ private[netty4] class SslClientVerificationHandler(
               )
             } else {
               val session = sslHandler.engine().getSession
-              verifySession(session, ctx)
-
-              if (onHandshakeComplete.setDone()) {
-                // If the original promise has not yet been satisfied, it can now be marked as successful,
-                // and the `SslClientVerificationHandler` can be removed from the pipeline. If the promise is
-                // already satisfied, the connection was failed and the failure has already been
-                // propagated to the dispatcher so we don't need to worry about cleaning up the pipeline.
-                ctx.pipeline.remove(self) // drains pending writes when removed
+              verifySession(session, ctx).onSuccess { _ =>
```

Comment:
I think this will just hang if it fails now?  can you add a test and make sure that's not the case?  we should do some kind of connection clean-up or error propagation on failure.  same on the server-side.